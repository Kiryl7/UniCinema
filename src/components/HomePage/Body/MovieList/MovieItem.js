import React from 'react'
import styles from './MovieList.css'


function MovieItem(props) {

    return (
            <li className={styles.indent}>
                <img src={props.data.img} className={styles.imgStyle} onClick={() => props.onOpen(props.data.id)} />
                <div className={styles.description}>
                    <b className={styles.rightB}>{props.data.title}</b>
                    <p>{props.data.age}</p>
                </div>
                <p>{props.data.ganre}</p>
            </li>
    )
}

export default MovieItem