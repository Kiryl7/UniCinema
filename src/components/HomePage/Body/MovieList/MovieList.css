.movieList {
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    list-style-type: none;
}

.indent {
    display: flex;
    flex-direction: column;
    align-items: center;
    margin: 1%;
}

.indent1 {
    padding: 1%;
}

.description { 
    display: flex;
}

.rightB{
    margin-right: 200px;
}

.imgStyle{
    height: 450px;
    width: 300px;
}
