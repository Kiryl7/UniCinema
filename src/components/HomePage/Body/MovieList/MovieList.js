import React from 'react'
import styles from './MovieList.css'

function MovieList() {
    return (
        <div className={styles.indent1}>
            <ul className={styles.movieList}>
                <li className={styles.indent}>
                    <img src="../../../../../images/topgun.jpg" width="300" height="400" alt="Top Gun"/>
                    <div className={styles.description}>
                        <b className={styles.rightB}>Top Gun</b>
                        <p>1994</p>
                    </div>                 
                    <p>Action</p>
                </li>
                <li className={styles.indent}>
                    <img src="../../../../../images/topgun.jpg" width="300" height="400" alt="Top Gun"/>
                    <div className={styles.description}>
                        <b className={styles.rightB}>Top Gun</b>
                        <p>1994</p>
                    </div>
                    <p>Action</p>
                </li>
                <li className={styles.indent}>
                    <img src="../../../../../images/topgun.jpg" width="300" height="400" alt="Top Gun"/>
                    <div className={styles.description}>
                        <b className={styles.rightB}>Top Gun</b>
                        <p>1994</p>
                    </div>
                    <p>Action</p>
                </li>
                <li className={styles.indent}>
                    <img src="../../../../../images/topgun.jpg" width="300" height="400" alt="Top Gun"/>
                    <div className={styles.description}>
                        <b className={styles.rightB}>Top Gun</b>
                        <p>1994</p>
                    </div>
                    <p>Action</p>
                </li>
                <li className={styles.indent}>
                    <img src="../../../../../images/topgun.jpg" width="300" height="400" alt="Top Gun"/>
                    <div className={styles.description}>
                        <b className={styles.rightB}>Top Gun</b>
                        <p>1994</p>
                    </div>
                    <p>Action</p>
                    </li>
                <li className={styles.indent}>
                    <img src="../../../../../images/topgun.jpg" width="300" height="400" alt="Top Gun"/>
                    <div className={styles.description}>
                        <b className={styles.rightB}>Top Gun</b>
                        <p>1994</p>
                    </div>
                    <p>Action</p>
                </li>
            </ul>
        </div>
    )
}

export default MovieList