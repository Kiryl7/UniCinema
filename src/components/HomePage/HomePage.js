import React from 'react'
import Header from './Header/Header.js'
import MidleContainer from './MidleContainer/MidleContainer.js'

import './HomePage.css'

function HomePage() {
    return (
        <div>
             <Header/> <MidleContainer/>
        </div>
    )
}

export default HomePage