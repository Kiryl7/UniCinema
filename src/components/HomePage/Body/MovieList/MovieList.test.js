import React from 'react';
import { render, screen, fireEvent } from "@testing-library/react"
import MovieList, { array as movies } from "./MovieList.js"
import renderer from 'react-test-renderer'
import HomePage from '../../HomePage.js';

it ("calls onClick when clicking on the movie", () => {
    const onClick = jest.fn();

    render(<MovieList onClick={onClick}/>);

    for (const movie of movies) {
        screen.getByText(movie.title);
    }

    const secondMovie = movies[1];
    fireEvent.click(screen.getByText(secondMovie.title));

    expect(onClick).toHaveBeenCalledTimes(1);
    expect(onClick).toHaveBeenCalledWith(secondMovie);
});

//jest --updateSnapshot

it("render correctly MovieList component", () => {
    const MovieListComponent = renderer.create(<MovieList/>).toJSON()
    expect(MovieListComponent).toMatchSnapshot()
});

it("render correctly HomePage component", () => {
    const MovieListComponent = renderer.create(<HomePage/>).toJSON()
    expect(MovieListComponent).toMatchSnapshot()
});
