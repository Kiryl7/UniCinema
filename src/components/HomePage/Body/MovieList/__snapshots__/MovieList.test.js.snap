// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly HomePage component 1`] = `
<div>
  <div
    className="tailTop"
  >
    <p
      className="header"
    >
      FIND YOUR MOVIE
    </p>
    <div
      className="search"
    >
      <div
        className="inputSearch"
      >
        <input
          className="size"
        />
      </div>
       
      <div
        className="buttonComponent"
      >
        <button>
          Search
        </button>
      </div>
    </div>
    <div
      className="searchFilter"
    >
      <p>
        SEARCH BY     
      </p>
      <div>
        <div
          className="form_toggle"
        >
          <div
            className="form_toggle_item item_1"
          >
            <input
              defaultChecked={true}
              id="fid_1"
              name="radio"
              type="radio"
              value="off"
            />
            <label
              htmlFor="fid_1"
            >
              TITLE
            </label>
          </div>
          <div
            className="form_toggle_item item_2"
          >
            <input
              id="fid_2"
              name="radio"
              type="radio"
              value="on"
            />
            <label
              htmlFor="fid_2"
            >
              GENRE
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
   
  <div
    className="midleLine"
  >
    <div
      className="resultsCount"
    >
      <p>
        X movie found
      </p>
    </div>
     
    <div
      className="resultsSort"
    >
      <div>
        <p>
          SORT BY     
        </p>
      </div>
      <div>
        <div
          className="form_toggle"
        >
          <div
            className="form_toggle_item item_1"
          >
            <input
              defaultChecked={true}
              id="fid_3"
              name="radio1"
              type="radio"
              value="off"
            />
            <label
              htmlFor="fid_3"
            >
              TITLE
            </label>
          </div>
          <div
            className="form_toggle_item item_2"
          >
            <input
              id="fid_4"
              name="radio1"
              type="radio"
              value="on"
            />
            <label
              htmlFor="fid_4"
            >
              GENRE
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
   
  <div
    className="body"
  >
    <div
      className="indent1"
    >
      <ul
        className="movieList"
      >
        <li
          className="indent"
          onClick={[Function]}
        >
          <img
            className="imgStyle"
            src="../../../../../images/topgun.jpg"
          />
          <div
            className="description"
          >
            <b
              className="rightB"
            >
              Top Gun
            </b>
            <p>
              1994
            </p>
          </div>
          <p>
            Action
          </p>
        </li>
        <li
          className="indent"
          onClick={[Function]}
        >
          <img
            className="imgStyle"
            src="../../../../../images/terminator.jpg"
          />
          <div
            className="description"
          >
            <b
              className="rightB"
            >
              Terminator
            </b>
            <p>
              1984
            </p>
          </div>
          <p>
            Science fiction/Action
          </p>
        </li>
        <li
          className="indent"
          onClick={[Function]}
        >
          <img
            className="imgStyle"
            src="../../../../../images/Blue_Velvet.jpg"
          />
          <div
            className="description"
          >
            <b
              className="rightB"
            >
              Blue Velvet
            </b>
            <p>
              1986
            </p>
          </div>
          <p>
            Detective film/Thriller
          </p>
        </li>
        <li
          className="indent"
          onClick={[Function]}
        >
          <img
            className="imgStyle"
            src="../../../../../images/Vanishing_Point.jfif"
          />
          <div
            className="description"
          >
            <b
              className="rightB"
            >
              Vanishing Point
            </b>
            <p>
              1971
            </p>
          </div>
          <p>
            Action / Cult Movie
          </p>
        </li>
        <li
          className="indent"
          onClick={[Function]}
        >
          <img
            className="imgStyle"
            src="../../../../../images/Death_Proof.jpg"
          />
          <div
            className="description"
          >
            <b
              className="rightB"
            >
              Death Proof
            </b>
            <p>
              2007
            </p>
          </div>
          <p>
            Action/Horror
          </p>
        </li>
        <li
          className="indent"
          onClick={[Function]}
        >
          <img
            className="imgStyle"
            src="../../../../../images/Lincoln_Lawyer.jpg"
          />
          <div
            className="description"
          >
            <b
              className="rightB"
            >
              Lincoln Lawyer
            </b>
            <p>
              2011
            </p>
          </div>
          <p>
            Thriller/Crime
          </p>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`render correctly MovieList component 1`] = `
<div
  className="indent1"
>
  <ul
    className="movieList"
  >
    <li
      className="indent"
      onClick={[Function]}
    >
      <img
        className="imgStyle"
        src="../../../../../images/topgun.jpg"
      />
      <div
        className="description"
      >
        <b
          className="rightB"
        >
          Top Gun
        </b>
        <p>
          1994
        </p>
      </div>
      <p>
        Action
      </p>
    </li>
    <li
      className="indent"
      onClick={[Function]}
    >
      <img
        className="imgStyle"
        src="../../../../../images/terminator.jpg"
      />
      <div
        className="description"
      >
        <b
          className="rightB"
        >
          Terminator
        </b>
        <p>
          1984
        </p>
      </div>
      <p>
        Science fiction/Action
      </p>
    </li>
    <li
      className="indent"
      onClick={[Function]}
    >
      <img
        className="imgStyle"
        src="../../../../../images/Blue_Velvet.jpg"
      />
      <div
        className="description"
      >
        <b
          className="rightB"
        >
          Blue Velvet
        </b>
        <p>
          1986
        </p>
      </div>
      <p>
        Detective film/Thriller
      </p>
    </li>
    <li
      className="indent"
      onClick={[Function]}
    >
      <img
        className="imgStyle"
        src="../../../../../images/Vanishing_Point.jfif"
      />
      <div
        className="description"
      >
        <b
          className="rightB"
        >
          Vanishing Point
        </b>
        <p>
          1971
        </p>
      </div>
      <p>
        Action / Cult Movie
      </p>
    </li>
    <li
      className="indent"
      onClick={[Function]}
    >
      <img
        className="imgStyle"
        src="../../../../../images/Death_Proof.jpg"
      />
      <div
        className="description"
      >
        <b
          className="rightB"
        >
          Death Proof
        </b>
        <p>
          2007
        </p>
      </div>
      <p>
        Action/Horror
      </p>
    </li>
    <li
      className="indent"
      onClick={[Function]}
    >
      <img
        className="imgStyle"
        src="../../../../../images/Lincoln_Lawyer.jpg"
      />
      <div
        className="description"
      >
        <b
          className="rightB"
        >
          Lincoln Lawyer
        </b>
        <p>
          2011
        </p>
      </div>
      <p>
        Thriller/Crime
      </p>
    </li>
  </ul>
</div>
`;
